packages/ui-extensions/src/surfaces/checkout/components/Abbreviation/Abbreviation.d.ts

```diff
@@ -0,0 +1,26 @@
+/** VERSION: 2025.7.0 **/
+/* eslint-disable import/extensions */
+/* eslint-disable @typescript-eslint/ban-types */
+/* eslint-disable @typescript-eslint/no-namespace */
+/* eslint-disable @typescript-eslint/member-ordering */
+/* eslint-disable line-comment-position */
+/* eslint-disable @typescript-eslint/unified-signatures */
+/* eslint-disable no-var */
+/* eslint-disable import/no-deprecated */
+/* eslint-disable import/namespace */
+/* eslint-disable import/no-deprecated */
+// eslint-disable-next-line @typescript-eslint/triple-slash-reference, spaced-comment
+/// <reference lib="DOM" />
+import type {AbbreviationProps$1} from './components-shared.d.ts';
+
+export interface AbbreviationProps extends Pick<AbbreviationProps$1, 'title'> {
+}
+export interface AbbreviationElement extends AbbreviationProps, Omit<HTMLElement, 'id' | 'title'> {
+}
+declare global {
```

Comment:
Just noticed that you're missing the preact jsx types. Also, we discovered that preact overwrites conflicting handler props that you have to omit (see related [PR](https://github.com/Shopify/ui-extensions/pull/2770) for Admin). Doing something like this should work
```ts
declare module 'preact' {
  namespace createElement.JSX {
    interface IntrinsicElements {
      's-abbreviation': Omit<
        HTMLAttributes<HTMLElement>,
        Extract<keyof HTMLAttributes<HTMLElement>, `on${Capitalize<string>}`>
      > &
      AbbreviationProps;
    }
  }
}
```

Reply:
I tophatted this, but the component types still aren't working and that is because of what Trish mentioned above ☝️ relating to the preact jsx types.

![Screenshot 2025-04-24 at 10 21 44 AM](https://github.com/user-attachments/assets/50f3f4d1-f14c-49b4-97a2-38b43a7281a6)